skbio/tree/_nj.py

```diff
@@ -216,15 +220,25 @@ def _lowest_index(dm):
     method (#228).
 
     """
-    lowest_value = np.inf
-    for i in range(dm.shape[0]):
-        for j in range(i):
-            curr_index = i, j
-            curr_value = dm[curr_index]
-            if curr_value < lowest_value:
-                lowest_value = curr_value
-                result = curr_index
-    return result
+    # convert to an array
+    dmarray = np.array(dm.to_data_frame())
+    # convert to dense, 1-d squareform of distance matrix
+    vec = squareform(dmarray)
+    # get the positions of the lowest value
+    results = np.vstack(np.where(dmarray == np.amin(vec))).T
+    # select results in the bottom-left of the array
+    results = [r for r in results if r[0] > r[1]]
+    # calculate the distances of the results to [0, 0]
+    res_distances = np.sqrt([r[0]**2 + r[1]**2 for r in results])
```

Comment:
We have other issues if the DMs are getting large enough for this to be a concern I believe. For instance, I don't think there is an architecture right now which has the memory necessary to support a 2^32 x 2^32 dense matrix. If you don't know about it, you can set the overflow behavior explicitly with numpy though:

```python
In [49]: np.geterr()   # np.seterr to change behavior
Out[49]: {'divide': 'warn', 'invalid': 'warn', 'over': 'warn', 'under': 'ignore'}
```